core-web/libs/sdk/experiments/src/lib/utils/utils.ts

```diff
@@ -0,0 +1,103 @@
+import { EXPERIMENT_ALLOWED_DATA_ATTRIBUTES, EXPERIMENT_SCRIPT_FILE_NAME } from '../contants';
+import { DataAttribute } from '../models';
+
+/**
+ * Returns the first script element that includes the experiment script identifier.
+ *
+ * @return {HTMLScriptElement|undefined} - The found script element or undefined if none is found.
+ */
+export const getExperimentScriptTag = (): HTMLScriptElement => {
+    const experimentScript = Array.from(document.getElementsByTagName('script')).find((script) =>
+        script.src.includes(EXPERIMENT_SCRIPT_FILE_NAME)
+    );
+
+    if (!experimentScript) {
+        throw new Error('Experiment script not found');
+    }
+
+    return experimentScript;
+};
+
+/**
+ * Retrieves experiment attributes from a given script element.
+ *
+ *
+ * @return {DataAttribute | null} - The experiment attributes or null if there are no valid attributes present.
+ */
+export const getDataExperimentAttributes = (): DataAttribute | null => {
+    const script = getExperimentScriptTag();
+    const defaultExperimentAttributes: DataAttribute = {
+        'api-key': '',
+        server: window.location.href,
+        debug: false
+    };
+
+    let experimentAttribute: Partial<DataAttribute> = {};
+
+    if (!script.hasAttribute('data-experiment-api-key')) {
+        Logger('You need specify the `data-experiment-api-key`');
+
+        return null;
+    }
```

Comment:
I'm not sure about how this works. But this means that the api key is embedded in the script tag? Is that safe?

Reply:
If so... that means that I can get that api-key, set this in another page and do a bunch of hit to "page view" and contaminate the experiment data?

Reply:
Yes rigth now that key with the current version of Experiment is like that.
**This is a running experiment in dotCMS:**
<img width="832" alt="Screenshot 2024-02-24 at 2 09 02 PM" src="https://github.com/dotCMS/core/assets/1909643/0ab502f5-094c-411e-9e90-24579f80a0e8">

Now, that key is validated and has a lifetime. But all this is validated by the backend.

Reply:
I talked with Falcon, the PO, and Will, the key is not a key, it is more like a token generated. By design right now, it is not necessary to hide that token.